File: myliquor-backend/src/main/java/memo/liquor/myliquor/entity/Review.java
package memo.liquor.myliquor.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Review {

    @Id @GeneratedValue
    @Column(name = "review_id")
    private Long id;

    private String content;

    private String imgName;

    private String imgPath;

    @ManyToOne
    @JoinColumn(name = "liquor_id")
    private Liquor liquor;

    public Review(String content, String imgName, String imgPath, Liquor liquor) {
        this.content = content;
        this.imgName = imgName;
        this.imgPath = imgPath;
        this.liquor = liquor;
    }
}

File: src/main/kotlin/record/liquors/liquor/entity/Review.kt
package record.liquors.liquor.entity

import javax.persistence.*

@Entity
class Review(
    @Lob
    var content: String,

    @Enumerated(EnumType.STRING)
    var rating: LiquorRating,

    @ManyToOne
    @JoinColumn(name = "liquor_id")
    var liquor: Liquor,

    @Id @GeneratedValue
    var id: Long? = null
) {
    fun changeContent(content: String) {
        this.content = content
    }
}